### PR TITLE
Add required option for function and assertions

### LINKED ISSUE #4: Adding require macro
Assert only checks the condition and goes ahead and runs all tests. but sometimes if function A doesn't work, then the rest of the tests will fail and we are sure about that.

We have to define `require` here. Example:

```c
NANO_REQUIRE_GE_INT( TESTNAME, expected, actual );
```

> Or maybe add a required field to asserts?

If the condition of these macros was false, then we will panic and halt. So, we won't wait a lot for tests that we are sure they will fail. it's very useful in pipelines and ...

### PR DESCRIPTION
**Pull request by kehiy**
- Fixes #4

### CODE REVIEW

File: src/lib.h [hunk 1]
```diff
@@ -88,6 +90,11 @@ static unsigned int TOTAL_SUCCESSFUL_COUNTER_PER_FUNCTION = 0;
 			TOTAL_FAILED_COUNTER_PER_FUNCTION++;                                                           \
 			fprintf( stderr, "- \t\"%s\" %s:%d Error: expected: %d, got: %d.\n", TESTNAME, __FILE__,       \
 				 __LINE__, expected, actual );                                                         \
+			if ( required )                                                                                \
+			{                                                                                              \
+				fprintf( stderr, "Test %s is required and must pass to continue.\n", TESTNAME );       \
```
Comment (0keeper1) [inline src/lib.h:95]: it's better to add a \tab before the line, like the one in the fprintf above ...

Comment (0keeper1) [reply src/lib.h:95]: But this line is not required ... you can handle It with one input in the print above ...

File: src/lib.h [hunk 2]
```diff
@@ -117,6 +125,11 @@ static unsigned int TOTAL_SUCCESSFUL_COUNTER_PER_FUNCTION = 0;
 			fprintf( stderr, "- \t\"%s\" %s:%d Error: expected %d to be greater than %d.\n", TESTNAME,     \
 				 __FILE__, __LINE__, expected, actual );                                               \
 		}                                                                                                      \
+		if ( required )                                                                                        \
```
Comment (0keeper1) [inline src/lib.h:128]: This condition just causes panic and doesn't need to be checked if the Assert passed.

```c
		if ( expected > actual )                                                                               \
		{                                                                                                      \
			TOTAL_FAILED_COUNTER++;                                                                        \
			TOTAL_FAILED_COUNTER_PER_FUNCTION++;                                                           \
			fprintf( stderr, "- \t\"%s\" %s:%d Error: expected %d to be greater than %d.\n", TESTNAME,     \
				 __FILE__, __LINE__, expected, actual );                                               \
			if ( required )                                                                                        \
		        {                                                                                                      \
			      fprintf( stderr, "Test %s is required and must pass to continue.\n", TESTNAME );               \
			      exit( 1 );                                                                                     \
		         }                                                                                                      \
		}                                                                                                      \
		else                                                                                                   \
		{                                                                                                      \
			TOTAL_SUCCESSFUL_COUNTER++;                                                                    \
			TOTAL_SUCCESSFUL_COUNTER_PER_FUNCTION++;                                                       \
			fprintf( stderr, "+ \t\"%s\" %s:%d Ok.\n", TESTNAME, __FILE__, __LINE__ );                     \
		}                                                                                                      \
```

**Review by 0keeper1 — CHANGES_REQUESTED** [review]

File: src/lib.h [hunk 3]
```diff
@@ -74,8 +75,9 @@ static unsigned int TOTAL_SUCCESSFUL_COUNTER_PER_FUNCTION = 0;
  * @param TESTNAME A human-readable name to identify the test.
  * @param expected The value the we that we expect to receive.
  * @param actual   The value which we received.
+ * @param required Indicates whether test proccess should panic if this test didn't passed.
 **/
-#define NANO_ASSERT_EQ_INT( TESTNAME, expected, actual )                                                               \
+#define NANO_ASSERT_EQ_INT( TESTNAME, expected, actual, required )                                                     \
```
Comment (0keeper1) [inline src/lib.h:80]: set required as optional input:
`__VA_OPT__`

https://en.cppreference.com/w/cpp/preprocessor/replace

**Review by 0keeper1 — CHANGES_REQUESTED** [review]

File: src/lib.h (hunk 3 above)
Comment (kehiy) [reply src/lib.h:80]: I'm facing with an strange warning:

```
__VA_OPT__ can only appear in the expansion of a variadic macro.
```

File: src/lib.h [hunk 4]
```diff
@@ -88,13 +88,10 @@ static unsigned int TOTAL_SUCCESSFUL_COUNTER_PER_FUNCTION = 0;
...
-				 __LINE__, expected, actual );                                                         \
-			if ( required )                                                                                \
-			{                                                                                              \
-				fprintf( stderr, "Test %s is required and must pass to continue.\n", TESTNAME );       \
-				exit( 1 );                                                                             \
-			}                                                                                              \
+			fprintf( stderr, "- \t\"%s\" %s:%d Error: expected: %d, got: %d.\n%s", TESTNAME, __FILE__,     \
+				 __LINE__, expected, actual,                                                           \
+				 required ? "This test is required and must pass to continue.\n" : "" );               \
+			assert( required == false );                                                                   \
```
Comment (0keeper1) [inline src/lib.h:94]: Assert is not required here use `exit(1);` instead.

Comment (kehiy) [reply src/lib.h:94]: We must exit only when requierd is set to true! not always!

**Pull request merged**